docs/language/informal/reserve-await-and-yield.md

```diff
@@ -0,0 +1,244 @@
+sync typedef yield
+```
+
+The language designers initially chose to do this for a few reasons:
+
+*   **It's more flexible.** Some keywords are really useful as identifiers and
+    it would be annoying if they were unavailable. For example, Dart has a Set
+    class, so it suck if you couldn't name a variable `set`. Likewise, it's
+    pretty common to see methods named `get()`, fields named `part`, etc.
+
+*   **It makes interop easier.** If Dart reserves a word that is a usable
+    identifier in another language, it can make it harder to call a foreign
+    method with that name. Dart was initially focused entirely on the web, so it
+    does not reserve any words that aren't reserved in JavaScript.
+
+As the language evolved, a third reason arose:
+
+*   **It's backwards compatible.** Introducing a new reserved word after the
+    language is out in the wild breaks existing uses of that word as an
+    identifier. That's why most other languages grow contextual keywords over
+    time.
+
+There are some cons to contextual keywords, though:
+
+*   **They are harder for users to understand.** Most programmers have a notion
+    of "reserved words". They may not be familiar with contextual keywords. If
+    they see one used as a keyword, they may not realize it can also be used
+    as an identifier. Likewise, if they see it used as an identifier, they may
+    be confused when they later see it treated like a keyword by the language.
+
+*   **They are hard to support with simple parsers.** Many tools that work with
+    Dart code don't use complete parsers for various reasons. In particular,
+    syntax highlighters often don't do a full parse. Instead, they break the
+    code into a flat sequence of tokens and use only local information to decide
+    how to color each one. That means they don't have the nested tree structure
+    neeeded to tell if a word like `on` is appearing in a catch clause and acts
+    like a keyword or is used as an identifier elsewhere.
+
+    The parsing framework, Grammar-Kit, used by the IntelliJ plug-in has
+    problems handling them.
+
+*   **It's harder to produce good error messages around them.** When an
+    erroneous piece of code uses a contextual keyword, it may not be clear if
+    the author was trying to use it as a keyword, or an identifier. Usually, a
+    smart tool can guess correctly, but it's still a guess. The error message
+    usually has to be worded to handle both cases, which tends to make it
+    confusing.
+
+## `await` and `yield`
+
+The last two problems apply in particular to `await` and `yield`. Most
+contextual keywords can be disambiguated using very local context. `hide` is
+only a keyword when right inside an import directive. `of` is only a keyword if
+it immediately follows `part` at the top level.
+
+But `await` is a keyword if the surrounding method is declared `async`. That
+modifier may be very far away from the use of `await`. `yield` is similar for
+`sync` methods.
```

Comment:
-> `sync*` methods 
Or better -> generator methods
(that covers both `sync*` and `async*`).